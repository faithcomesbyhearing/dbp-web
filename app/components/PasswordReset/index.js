/**
 *
 * PasswordReset
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import PopupMessage from '../PopupMessage';
import checkEmail from '../../utils/checkEmailForValidity';

class PasswordReset extends React.PureComponent {
	constructor(props) {
		super(props);
		this.state = {
			email: '',
		};
	}

	handleSubmit = (e) => {
		e.preventDefault();
		const email = this.state.email;
		if (email && checkEmail(email)) {
			this.props.resetPassword(e, { email: this.state.email });
		} else {
			const client = e.target.childNodes[1].getBoundingClientRect() || {
				x: 0,
				y: 0,
			};

			this.props.openPopup({ x: client.x, y: client.y });
		}
	};

	handleInputChange = (e) => this.setState({ email: e.target.value });

	render() {
		const {
			popupCoords,
			popupOpen,
			message = 'Please enter a valid email.',
		} = this.props;
		const { email } = this.state;

		return (
			<>
				<div className="forgot-password">
					<p>
						In order to reset your password, please enter the email address you
						used to register for Bible.is.
					</p>
					<form onSubmit={this.handleSubmit} className={'wrapper'}>
						<input
							onChange={this.handleInputChange}
							value={email}
							autoComplete={'email'}
							placeholder="E-mail"
						/>
						<button type={'submit'} className="text">
							Reset Password
						</button>
					</form>
					<div className="disclaimer">
						If you are unable to reset your password, please
						<a href="https://support.bible.is/contact" className="link">
							&nbsp;contact us&nbsp;
						</a>
						for support.
					</div>
				</div>
				{popupOpen ? (
					<PopupMessage x={popupCoords.x} y={popupCoords.y} message={message} />
				) : null}
			</>
		);
	}
}

PasswordReset.propTypes = {
	popupOpen: PropTypes.bool,
	message: PropTypes.string,
	openPopup: PropTypes.func,
	resetPassword: PropTypes.func,
	popupCoords: PropTypes.object,
};

export default PasswordReset;
