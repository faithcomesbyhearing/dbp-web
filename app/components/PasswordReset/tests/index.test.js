import React from 'react';
import renderer from 'react-test-renderer';
import PasswordReset from '..';

const props = {
	popupOpen: false,
	message: 'Success!',
	openPopup: jest.fn(),
	resetPassword: jest.fn(),
	popupCoords: {
		x: 150,
		y: 150,
	},
};

describe('<PasswordReset /> Component', () => {
	it('Should match the previous snapshot', () => {
		const tree = renderer.create(<PasswordReset {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
