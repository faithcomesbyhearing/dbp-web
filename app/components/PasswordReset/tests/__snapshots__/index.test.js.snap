// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordReset /> Component Should match the previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="forgot-password"
  >
    <p>
      In order to reset your password, please enter the email address you used to register for Bible.is.
    </p>
    <form
      class="wrapper"
    >
      <input
        autocomplete="email"
        placeholder="E-mail"
        value=""
      />
      <button
        class="text"
        type="submit"
      >
        Reset Password
      </button>
    </form>
    <div
      class="disclaimer"
    >
      If you are unable to reset your password, please
      <a
        class="link"
        href="https://support.bible.is/contact"
      >
         contact us 
      </a>
      for support.
    </div>
  </div>
</DocumentFragment>
`;
